### PR TITLE
Internal: Elements Regression - Summary for tested controls [ED-8292]

### PR DESCRIPTION
**Pull request by matipojo**

### CODE REVIEW

File: tests/elements-regression/tests/elements-regression.test.js [hunk 1]
```diff
@@ -4,13 +4,16 @@ const elementsConfig = require( '../elements-config.json' );
...
 const controlHandlers = require( '../src/controls' );
+const { summary } = require( '../src/utils' );
 
 const configMediator = ConfigProvider.make( { elementsConfig, testConfig } );
 
 test.describe( 'Elements regression', () => {
 	const testedElements = {};
 
 	test.afterAll( async ( {}, testInfo ) => {
+		console.log( 'summaryData', summary( testedElements, elementsConfig ) );
```
Comment (Nevoss) [inline tests/elements-regression/tests/elements-regression.test.js:15]: Add disable lint comment

File: tests/elements-regression/src/utils.js [hunk 2]
```diff
@@ -1,3 +1,33 @@
...
+	 * @param {Object} tested   e.g. { heading: { title: 'Hello World' } }
+	 * @param {Object} expected e.g. { heading: { controls: { title: 'Hello World' } } }
+	 * @return {{
+	 * 	testedWidgetsCount: number,
+	 * 	expectedWidgetsCount: number,
+	 * 	testedControlsCount: number,
+	 * 	expectedControlsCount: number,
+	 * }}
+	 */
+	summary: ( tested, expected ) => {
```
Comment (Nevoss) [inline tests/elements-regression/src/utils.js:14]: Maybe we should use a custom reporter alongside the regular reporter?
https://playwright.dev/docs/test-reporters#custom-reporters

**Review by Nevoss — CHANGES_REQUESTED** [review]

File: tests/elements-regression/tests/elements-regression.test.js (hunk 1 above)
Comment (matipojo) [inline tests/elements-regression/tests/elements-regression.test.js:15]:
```suggestion
		// eslint-disable-next-line no-console
		console.log( 'summaryData', summary( testedElements, elementsConfig ) );
```

File: tests/elements-regression/tests/elements-regression.test.js [hunk 3]
```diff
@@ -4,13 +4,17 @@ const elementsConfig = require( '../elements-config.json' );
...
+const { summary } = require( '../src/utils' );
 
 const configMediator = ConfigProvider.make( { elementsConfig, testConfig } );
 
 test.describe( 'Elements regression', () => {
 	const testedElements = {};
 
 	test.afterAll( async ( {}, testInfo ) => {
+		// eslint-disable-next-line no-console
+		console.log( 'summaryData', summary( testedElements, elementsConfig ) );
```
Comment (matipojo) [inline tests/elements-regression/tests/elements-regression.test.js:16]:
```suggestion
		const summaryData = summary( testedElements, elementsConfig );
		test.info().annotations.push( { type: 'summary', description: JSON.stringify( summaryData ) } );
```

File: tests/elements-regression/tests/elements-regression.test.js [hunk 4]
```diff
@@ -4,13 +4,17 @@ const elementsConfig = require( '../elements-config.json' );
...
+const { summary } = require( '../src/utils' );
 
 const configMediator = ConfigProvider.make( { elementsConfig, testConfig } );
 
 test.describe( 'Elements regression', () => {
 	const testedElements = {};
 
 	test.afterAll( async ( {}, testInfo ) => {
+		const summaryData = summary( testedElements, elementsConfig );
+		test.info().annotations.push( { type: 'summary', description: JSON.stringify( summaryData ) } );
```
Comment (matipojo) [inline tests/elements-regression/tests/elements-regression.test.js:16]:
```suggestion
		// eslint-disable-next-line no-console
		console.log( 'summaryData', summary( testedElements, elementsConfig ) );
```

**Pull request merged**